pkg/boot/menu/menu.go

```diff
@@ -51,12 +52,9 @@ type Entry interface {
 }
 
 func parseBootNum(choice string, entries []Entry) (int, error) {
-	num, err := strconv.Atoi(choice)
-	if err != nil {
-		return -1, fmt.Errorf("%s is not a valid entry number: %v", choice, err)
-	}
-	if num < 1 || num > len(entries) {
-		return -1, fmt.Errorf("%s is not a valid entry number", choice)
+	num, err := strconv.Atoi(strings.TrimSpace(choice))
+	if err != nil || num < 1 || num > len(entries) {
+		return -1, fmt.Errorf("'%s' is not a valid entry number", choice)
```

Comment:
Instead of `'%s'` use `%q`

Reply:
Done.

Reply:
Actually is %q valid?

%q	a single-quoted character literal safely escaped with Go syntax.

choice isn't necessarily a single character. The purpose of the ' ' was to encapsulate the entire string in a displayable format, with spaces/tabs/etc included, would it be better to escape some "s?

Reply:
Ah, never-mind, I was misinterpreting 'character literal' to be a single character, it is not